Dockerfile

```diff
@@ -30,14 +37,25 @@ ENV NODE_ENV=production
 WORKDIR /home/node/app
 
 # Copy only necessary files from build stage
-COPY --from=build --chown=node:node /home/node/app/packages ./packages
-COPY --from=build  --chown=node:node /home/node/app/node_modules ./node_modules
+COPY --from=builder --chown=node:node /home/node/app/packages ./packages
+COPY --from=builder  --chown=node:node /home/node/app/node_modules ./node_modules
 
-# Switch to non-root user
-USER node
+# Base arguments: build-time
+ARG NPM_CONFIG_LOGLEVEL=warn
+ARG PORT=3000
+
+# Environment variables: base configuration
+ENV NPM_CONFIG_LOGLEVEL=${NPM_CONFIG_LOGLEVEL}
+ENV PORT=${PORT}
 
-# Expose port
-EXPOSE 3000
+# Set ownership permissions
+RUN chown -R node:node /home/node/app
+
+# Specify default port
+EXPOSE ${PORT}
+
+# Set user and shell
+USER node
 
-# Start the server
+# Run the application
 CMD ["node", "packages/server/build/index.js"]
```

Comment:
Define as `npm start` (see example in Studio) instead as then it can more generally the specific command can evolve

Reply:
Done